core/domain/topic_domain.py

```diff
@@ -601,47 +612,82 @@ def update_subtopic_title(self, subtopic_id, new_title):
                 'The subtopic with id %s does not exist.' % subtopic_id)
         self.subtopics[subtopic_index].title = new_title
 
-    def move_skill_id_to_subtopic(self, subtopic_id, new_skill_id):
-        """Updates the title of the new subtopic.
+    def move_skill_id_to_subtopic(
+            self, old_subtopic_id, new_subtopic_id, skill_id):
+        """Moves the skill_id to a new subtopic or to uncategorized skill ids.
 
         Args:
-            subtopic_id: str. The id of the subtopic to which the skill is to
-                be moved.
-            new_skill_id: str. The new skill id which is to be added to the
-                subtopic.
+            old_subtopic_id: str or None. The id of the subtopic in which the
+                skill is present currently (before moving) or None is it is
+                uncategorized.
+            new_subtopic_id: str. The new subtopic to which the skill is
+                to be moved.
+            skill_id: str. The skill id which is to be moved.
 
         Raises:
             Exception. The subtopic with the given id doesn't exist.
-            Exception. The skill id should be present in the topic already
-                before moving.
+            Exception. The skill id should be present in the old subtopic
+                (or uncategorized) already before moving.
             Exception. The skill id is already present in the subtopic.
         """
+        if old_subtopic_id is not None:
+            old_subtopic_index = self.get_subtopic_index(old_subtopic_id)
+            if old_subtopic_index is None:
+                raise Exception(
+                    'The subtopic with id %s does not exist.' % old_subtopic_id)
+            if skill_id not in self.subtopics[old_subtopic_index].skill_ids:
+                raise Exception(
+                    'Skill id %s is not present in the given old subtopic'
+                    % skill_id)
+
+        if old_subtopic_id is None:
```

Comment:
Do "else" instead. Cleaner since it's obvious to the reader that only one branch will execute.

Reply:
Done.